app/src/main/java/com/example/concordia_campus_guide/Activities/PathsActivity.java

```diff
@@ -0,0 +1,139 @@
+import android.widget.ImageButton;
+import android.widget.TextView;
+
+import com.example.concordia_campus_guide.Adapters.DirectionWrapper;
+import com.example.concordia_campus_guide.Fragments.LocationFragment.LocationFragment;
+import com.example.concordia_campus_guide.Fragments.PathInfoCardFragment.PathInfoCardFragment;
+import com.example.concordia_campus_guide.GoogleMapsServicesTools.GoogleMapsServicesModels.DirectionsResult;
+import com.example.concordia_campus_guide.GoogleMapsServicesTools.GoogleMapsServicesModels.DirectionsStep;
+import com.example.concordia_campus_guide.Models.Place;
+import com.example.concordia_campus_guide.Models.RoomModel;
+import com.example.concordia_campus_guide.Models.WalkingPoint;
+import com.example.concordia_campus_guide.R;
+import com.google.android.material.bottomsheet.BottomSheetBehavior;
+import java.util.ArrayList;
+import java.util.List;
+
+public class PathsActivity extends AppCompatActivity {
+    PathsViewModel mViewModel;
+    LocationFragment locationFragment;
+    TextView fromTextView;
+    TextView toTextView;
+    ImageButton backButton;
+    PathInfoCardFragment pathInfoCardFragment;
+    FragmentTransaction fragmentTransaction;
+    FragmentManager fragmentManager;
+    private BottomSheetBehavior swipeableInfoCard;
+    private DirectionsResult directionsResult;
+    private ArrayList<DirectionWrapper> directionWrappers;
+    Place from;
+    Place to;
+    private boolean fromIsIndoor = false;
+    private boolean toIsIndoor = false;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.paths_activity);
+
+        fromTextView = (TextView) findViewById(R.id.path_fromText);
+        toTextView = (TextView) findViewById(R.id.path_toText);
+
+        mViewModel = ViewModelProviders.of(this).get(PathsViewModel.class);
+        fragmentManager = getSupportFragmentManager();
+
+        Bundle extras = getIntent().getExtras();
+        if (extras != null) {
+            directionsResult = (DirectionsResult) extras.getSerializable("directionsResult");
+        }
+
+        locationFragment = (LocationFragment) getSupportFragmentManager().findFragmentById(R.id.pathLocationFragment);
+
+        from = mViewModel.getFrom();
+        to = mViewModel.getTo();
+        checkFromToType(from, to);
+
+        fromTextView.setText(from.getDisplayName());
+        toTextView.setText(to.getDisplayName());
+        setBackButtonOnClickListener();
+
+        directionWrappers = !(fromIsIndoor && toIsIndoor) ? (ArrayList<DirectionWrapper>) parseDirectionResults() : new ArrayList<DirectionWrapper>();
+        if (!(fromIsIndoor && toIsIndoor)) {
+            locationFragment.drawOutdoorPaths(directionWrappers);
+        }
+
+        View pathInfoCard = findViewById(R.id.path_info_card);
+        swipeableInfoCard = BottomSheetBehavior.from(pathInfoCard);
+        setIndoorPaths();
+        showInfoCard();
```

Comment:
You should refactor the code here, by creating a method that would take care of the initComponent() for example.